### DISCUSSION TITLE
Form Validation with attributes

### DISCUSSION THREAD

**Discussion by danielraab** [livewire/livewire#7030] [7. Bugs (archived)]
### Livewire version

v3.0.6

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

I am not sure if its a bug or a documentation issue.

1. Use a Form in a Livewire-Component with some Validaton Rules (the kind of applying the rules does not matter):
```php
            'entrance_date' => ['required', 'date'],
            'leaving_date' => ['nullable', 'date', 'after_or_equal:entrance_date'],
```
3. This Form contains a store function where in the validation function is called (`$this->validate()`).
4. If the leaving_date contains a date before the entrance_date i would assume to get an ValidationException and an error displayed. But i get no error.

### Additional information

The reason is that the \Livewire\Features\SupportFormObjects\Form calles the "parent components getRules() function:
```php
        $rules = $this->component->getRules();
        foreach ($rules as $key => $value) {
            if (! str($key)->startsWith($this->propertyName . '.')) continue;

            $filteredRules[$key] = $value;
        }
```
This impacts in an prefix for the Rule Key: e.g. `memberForm.entrance_date`.  All other found rules (from the parent component) will be filtered out and are not validated). So if i define this rules, everything works as expected and i get my ValidationException:
```php
            'entrance_date' => ['required', 'date'],
            'leaving_date' => ['nullable', 'date', 'after_or_equal:memberForm.entrance_date'],
```

I think the behavior that the validation keys are prefixed with the property name of the form is a little bit misleading. The reusability of the Form Object suffers too. On every use of it, it has to have the same property name. Otherwise the rule is not working.

At leased it is not documented (i was not able find anything in the documentation).

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment]
@danielraab have a look at PR #6614 🙂
